cmd/cloudFoundryDeploy.go

```diff
@@ -668,21 +672,101 @@ func deployMta(config *cloudFoundryDeployOptions, mtarFilePath string, command c
 		cfDeployParams = append(cfDeployParams, deployParams...)
 	}
 
-	cfDeployParams = append(cfDeployParams, handleMtaExtensionDescriptors(config.MtaExtensionDescriptor)...)
+	extFileParams, extFiles := handleMtaExtensionDescriptors(config.MtaExtensionDescriptor)
 
-	return cfDeploy(config, cfDeployParams, nil, nil, command)
+	for _, extFile := range extFiles {
+		_, err := fileUtils.Copy(extFile, extFile+".original")
+		if err != nil {
+			return err
+		}
+		err = handleMtaExtensionCredentials(extFile, config.MtaExtensionCredentials)
+		if err != nil {
+			return err
```

Comment:
From looking at the implementation, it can return an error that is lacking any context. So should be wrapped as well.

Reply:
done